### ISSUE TITLE
Please move static analysis dependencies in compileOnly scope

### ISSUE THREAD

**Issue by remal** [testcontainers/testcontainers-java#1824]
Currently these dependencies are in `compile` scope:
* `org.jetbrains:annotations`
* `javax.annotation:javax.annotation-api`

Please move them into `compileOnly`. I suppose Testcontainers users don't need these libraries in classpath.

**Comment by bsideup** [comment]
Alternatively, we could have our own annotations since JSR-305 allows having them.

Here is an example from Project Reactor:
https://github.com/reactor/reactor-core/tree/master/reactor-core/src/main/java/reactor/util/annotation

**Comment by remal** [comment]
@bsideup Yes, it's possible. But I'd prefer having a limited number of `NonNullApi` annotations in the project's classpath. For example, Spring Framework has its own `NonNullApi` annotations.

BTW using `compileOnly` scope won't break anything.

**Comment by bsideup** [comment]
The whole idea of `@TypeQualifierNickname` is to be able to have project-specific annotations, so that there is no need to depend on a common dependency with the annotations. This is why there are such annotations in Spring and Project Reactor.

Also, they are not supposed to be used by the end user, but only to provide info about the existing code of the project, so the amount of the annotations on project's classpath should not be a concern.

**Comment by remal** [comment]
I have nothing against the idea. But as I've seen all these project-specific annotations don't work if `javax.annotation.*` annotations are not presented in the classpath.

**Comment by remal** [comment]
And... I think that annotations' scope and NonNullApi annotations are different things. I'd like to repeat: everything will continue to work with `compileOnly` scope.

**Comment by MariuszCwikla** [comment]
I'd like to pick up this issue.
I made some observations already:
- I think that `javax.annotation:javax.annotation-api` could be removed. I didn't see any usages
- `org.jetbrains:annotations` - moving `compileOnly` from `core` to `subprojects/dependencies` should fix it (since other modules also use these annotations)
